File: book1_companion/psych_stats_with_python_companion_v0_1/scripts/python/ch11_regression.py
#!/usr/bin/env python3
from __future__ import annotations

import pandas as pd
from scipy import stats

from common import ROOT, rounded, write_result


def main() -> None:
    data_file = "data/ch11_regression.csv"
    data = pd.read_csv(ROOT / data_file)
    result = stats.linregress(data["study_hours"], data["test_score"])
    fields = {
        "intercept": rounded(result.intercept),
        "slope": rounded(result.slope),
        "slope_standard_error": rounded(result.stderr),
        "t_statistic": rounded(result.slope / result.stderr),
        "degrees_of_freedom": int(len(data) - 2),
        "p_value": rounded(result.pvalue),
        "r_squared": rounded(result.rvalue**2),
        "n": int(len(data)),
    }
    out = write_result("ch11", "Simple linear regression", data_file, fields)
    print(f"wrote {out}")


if __name__ == "__main__":
    main()

File: book1_companion/psych_stats_with_python_companion_v0_2/scripts/r/ch11_regression.R
#!/usr/bin/env Rscript
args <- commandArgs(trailingOnly = FALSE)
script_arg <- args[grep('^--file=', args)][1]
root <- normalizePath(file.path(dirname(sub('^--file=', '', script_arg)), '..', '..'), mustWork = TRUE)
source(file.path(root, "scripts", "r", "common.R"))

d <- read.csv(file.path(root, "data", "ch11_regression.csv"))
fit <- lm(test_score ~ study_hours, data = d)
s <- summary(fit)
coefs <- s$coefficients
write_result(root, "ch11", "Simple linear regression", "data/ch11_regression.csv", list(
  intercept = coefs["(Intercept)", "Estimate"], slope = coefs["study_hours", "Estimate"],
  slope_standard_error = coefs["study_hours", "Std. Error"], t_statistic = coefs["study_hours", "t value"],
  degrees_of_freedom = s$df[2], p_value = coefs["study_hours", "Pr(>|t|)"], r_squared = s$r.squared, n = nrow(d)
))
